contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,302 @@
+# Scheduling Policy
+
+_Status: Draft_
+
+_Authors: @arnaudmz, @yastij_
+
+_Reviewers: @bsalamat, @tallclair, @liggitt_
+
+# Objectives
+
+-  Define the concept of scheduling policies
+-  Propose their initial design and scope
+
+## Non-Goals
+
+-  How taints / tolerations work.
+-  How NodeSelector works.
+-  How node / pod affinity / anti-affinity rules work.
+-  How several schedulers can be used within a single cluster.
+-  How priority classes work.
+-  How to set defaults in Kubernetes.
+
+# Background
+
+During real-life Kubernetes architecturing we encountered contexts where role-isolation (between administration and simple namespace usage in a multi-tenant context) could be improved. So far, no restriction is possible on toleration, priority class usage, nodeSelector, anti-affinity depending on user permissions (RBAC).
+
+Identified use-cases aim to ensure that administrators have a way to restrict users or namespaces, it allows administrators to:
+
+-  Restrict execution for specific applications (which are namespace scoped) into certain nodes
+-  Create policies that prevent users from even attempting to schedule workloads onto masters to maximize security
+-  require that a pods under a namespace run on dedicated nodes
+-  Restrict usage of some `PriorityClass`
+-  Restrict usage to a specific set of schedulers.
+-  enforcing pod affinity or anti-affinity rules on some particular namespace.
+
+Also Mandatory values must also be enforced by scheduling policies in case of:
+
+-  mutli-arch (amd64, arm64) of multi-os (Linux, Windows) clusters (will also be handled later by the [RuntimeClass]() KEP)
+-  multi-az / region / failure domain clusters
+
+# Overview
+
+### syntaxic standards
+
+### empty match and unset fields:
+
+We follow Kubernetes conventions, an empty field matches everything an unset one matches nothing
+
+### how policies are computed
+
+Policies are computed using the following algorithm:
+
+```
+sortedPolicies = sort_by_priority(sort_deny_first(policies))
+for policy in sortedPolicies:
+  if policy matches pod: // all specified policy rules match
+    return policy.action
+```
+ note that rules of policies from a lower priority are superseeded by ones from a higher priority if they match.
+
+
+
+# Detailed Design
+
+
+## SchedulingPolicy
+
+Proposed API group: `policy/v1alpha1`
+
+
+### SchedulingPolicy content
+
+SchedulingPolicy spec is composed of optional fields that allow scheduling rules. If a field is absent from a SchedulingPolicy, this `SchedulingPolicy` won't allow any item from the missing field.
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  priority: <exception,cluster,default>
+  action:  <allow,deny>     
+  rules:
```

Comment:
nit: I don't love the name `rules` for this field (I'd expect it to be of `[]Rule` type), but I don't have any good suggestions.... maybe `criteria`?

Alternatively we could just flatten it and put the rules directly in the spec.